posts/2022-08-04-jakarta-core-profile-22009-beta.adoc

```diff
@@ -0,0 +1,270 @@
+
+
+[#json]
+== Jakarta JSON Binding and Processing
+The Jakarta JSON Processing specification allows users to parse and generate JSON strings. The Jakarta JSON Binding specification is built on top of JSON Processing and allows users to serialize and deserialize Java objects into JSON.
+
+The Jakarta JSON Processing specification has been updated with enhancements for handling duplicate keys when parsing JSON, and better handling of `java.lang.Number` superclass when generating JSON strings. The Jakarta JSON Binding specification has been updated to support polymorphism. 
+
+Open Liberty provides default implementations for both Jakarta JSON Processing and Binding. For JSON Processing we provide users with the Parsson implementation. For JSON Binding we provide users with the Yasson implementation.
+To get these default implementations you can enable the `jsonp-2.1` and/or `jsonb-3.0`` features.
+
+[source, xml]
+----
+<featureManager>
+    <feature>jsonb-3.0</feature>
+    <feature>jsonp-2.1</feature>
+</featureManager>
+----
+
+If you want to use a third-party Jakarta JSON Processing 2.1 or Binding 3.0 implementation you can do so by enabling the `jsonpContainer-2.1` and `jsonbContainer-3.0` features respectively. The JSON Container features require users to configure a Basic Extensions using Liberty Libraries (BELLs) element that points to their third-party implementation(s).
+
+[source, xml]
+----
+<featureManager>
+    <feature>jsonbContainer-3.0</feature>
+    <feature>jsonpContainer-2.1</feature>
+</featureManager>
+
+<library id="myJsonBindingProvider">
+  <file name="${path.to.json.binding.provider}"/>
+</library>
+
+<bell libraryRef="myJsonBindingProvider"/>
+
+<library id="myJsonProcessingProvider">
+  <file name="${path.to.json.processing.provider}"/>
+</library>
+
+<bell libraryRef="myJsonProcessingProvider"/>
+----
+
+NOTE: When you enable `jsonb-3.0` or `jsonbContainer-3.0` the `jsonp-2.1` feature is automatically enabled. However, enabling the `jsonpContainer-2.1` feature will supersede the `jsonp-2.1` feature and allow the user to provide a third-party implementation.
+
+You can find out more about this specific update (eg Open Liberty docs, Javadoc) and/or the wider technology via the following links:
+
+* link:{url-prefix}/docs/latest/json-p-b.html[Open Liberty JSON-P and JSON-B Documentation]
+* link:https://javaee.github.io/jsonp/getting-started.html[Jakarta JSON Processing - Getting Started]
+* link:https://javaee.github.io/jsonb-spec/getting-started.html[Jakarta JSON Binding - Getting Started]
+* link:{url-prefix}/guides/rest-client-java.html[Open Liberty REST Client Guide]
+
+
+[#cdi]
+== CDI 4.0
+CDI allows objects to be bound to lifecycle contexts, to be injected, to be associated with interceptors and decorators, and to interact in a loosely coupled fashion by firing and observing events.
```

Comment:
This whole section (under the CDI 4.0 heading) needs work. The bulleted list format is too long to read and make sense of. This blog post isn't just release notes for existing developers but needs to be appealing to new developers so that they can see how great CDI is and what it can do for them, and why they should try using Jakarta EE 10/MicroProfile at all.

In the introductory para, can we explain in a sentence each what the Annotations and Interceptors APIs provide (for someone new to CDI)?

Remove "As part of Jakarta EE 10 Core Profile," as it doesn't really add anything to that sentence.

"brings some functional updates" doesn't really tell us anything - what are the really key changes in this release that are likely to be of interest to the most developers? Surely all the changes aren't really minimal and equally mundane?

What's "Build Compatible Extensions" and "a new Metamodel" mean? What benefit do these changes have to a developer? And why is that change different from the bulleted list below it - is it more important?

Can the updates that are currently in the long bulleted list be grouped in some way and then each group have at least an introductory couple of sentences explaining why the changes have been made at a higher level? For example, "The following APIs were deprecated in previous releases of Open Liberty and have now been removed:.....list of changes here....If your application uses these APIs, it will xxxx so you need to update them to xxx."

Maybe use the words/groupings from the original bulleted list at the top of the post? "defining new extension APIs; removing reflections; adding new reflection-free language model; empty beans.xml means that only annotated classes are beans; removal of deprecated APIs".

Reply:
@tevans78 ^

Reply:
https://github.com/OpenLiberty/blogs/pull/2427